### DISCUSSION TITLE
Time since First Dose

### DISCUSSION THREAD

**Discussion by Ravua1992** [metrumresearchgroup/mrgsolve#1122] [Q&A] [answered]
Hello, 

How do I get time since first dose in the model ? 
I am aware of TAD which is `capture TAD= self.tad();` similarly, I am looking for something which can capture Time since first and last dose?
TSFD and TSLD. 

Thank you!

**Comment by kylebaron** [comment, marked as answer]
Hi @Ravua1992  - 

There isn't anything baked into mrgsolve to do time since first dose; but here's some code if you did want to derived it inside the model. This might be more easily done after you simulate but you can do it in the model too. 

Kyle

``` r
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

code <- '
$PKMODEL cmt = "CENT"

$PARAM CL = 1, V = 20

$PK
if(NEWIND <=1) double tofd = -1;

if(EVID == 1 && tofd < 0) tofd = TIME; 

$ERROR
capture TAFD = tofd >= 0 ? TIME - tofd : 0;
'

mod <- mcode("tafd", code)
#> Building tafd ...
#> done.

mod %>% 
  ev(amt = 100, time = 12, ii = 24, addl = 2) %>% 
  mrgsim(end = 144) %>% 
  plot()
```

![](https://i.imgur.com/vYxF3ar.png)

<sup>Created on 2023-10-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by Ravua1992** [reply]
Thank you!
